src/scripts/Update.ts

```diff
@@ -598,6 +599,47 @@ class Update implements Saveable {
             saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 32, 34); // New Mauville
             saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 84, 64); // Pledge Grove
             saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 78, 79); // Abundant Shrine
+
+            // Shards -> Gems
+            delete saveData.quests.questList;
+
+            //Setting gems = shards
+            saveData.gems = saveData.shards || {};
+            saveData.gems = {
+                ...saveData.gems,
+                gemWallet: saveData.shards.shardWallet || 0,
+                gemCollapsed: saveData.shards.shardCollapsed || 0,
+                gemUpgrades: saveData.shards.shardUpgrades || 0,
+            };
+
+            delete saveData.keyItems['Shard_case'];
+
+            // Swapping Shard Case for Gem Case
+            if (saveData.badgeCase[8]) {
+                saveData.keyItems['Gem_case'] = true;
+            }
+
+            // Just incase statistics is not set
+            saveData.statistics = saveData.statistics || {};
+
+            // Rename from the old statistic name
+            saveData.statistics = {
+                ...saveData.statistics,
+                totalGemsGained: saveData.statistics.totalShardsGained || 0,
+                gemsGained: saveData.statistics.shardsGained || 0,
+            };
+
+            // Disable Gems Challenge, this is kind of a crappy solution but eh
+            if (saveData.gems.gemUpgrades.every((g: number) => !g)) {
+                Notifier.notify({
+                    title: 'Active Challenge Mode?',
+                    message: `Do you want to activate the No Gem challenge mode?
+                    (Synonymous with the former "No Shard" Challenge)
+
+                    <button class="btn btn-block btn-danger" onclick="App.game.challenges.list.disableGems.activate();" data-dismiss="toast">Activate</button>`,
+                    timeout: GameConstants.HOUR,
+                });
+            }
```

Comment:
Can't we just check if the shard challenge is enabled/disabled and use the same value?

Reply:
This _should_ work (again untested):
```js
saveData.challenges.list.disableGems = saveData.challenges?.list?.disableShards || false;
```

Reply:
Works great!